tasks/phylogenetic_inference/task_lyveset.wdl

```diff
@@ -0,0 +1,146 @@
+    ##PERFORM CERTAIN STEPS
+    ##--mask-phages                  Search for and mask phages in the reference genome
+    ##--mask-cliffs                  Search for and mask 'Cliffs' in pileups
+    ##
+    ##SKIP CERTAIN STEPS
+    ##--nomatrix                     Do not create an hqSNP matrix
+    ##--nomsa                        Do not make a multiple sequence alignment
+    ##--notrees                      Do not make phylogenies
+    ##--singleend                    Treat everything like single-end. Useful
+    ##                               for when you think there is a single-
+    ##                               end/paired-end bias.
+    ##OTHER SHORTCUTS
+    ##--fast                         Shorthand for --downsample --mapper snap --nomask-phages
+    ##                                             --nomask-cliffs --sample-sites
+    ##--downsample                   Downsample all reads to 50x. Approximated according
+    ##                               to the ref genome assembly
+    ##--sample-sites                 Randomly choose a genome and find SNPs in a quick
+    ##                               and dirty way. Then on the SNP-calling stage,
+    ##                               only interrogate those sites for SNPs for each
+    ##                               genome (including the randomly-sampled genome).
+    ##
+    ##MODULES
+    ##--read_cleaner none            Which read cleaner? Choices: none, CGP, BayesHammer
+    ##--mapper       smalt           Which mapper? Choices: smalt, snap
+    ##--snpcaller    varscan         Which SNP caller? Choices: varscan, vcftools
+    Int allowedFlanking = 0
+    Float min_alt_frac = 0.75
+    Int min_coverage = 10
+    String? presets
+    Boolean mask_phages = false
+    Boolean mask_cliffs = false
+    Boolean nomatrix = false
+    Boolean nomsa = false
+    Boolean notrees = false
+    # Boolean singleend = false -- currently written for PE read data only
+    Boolean fast = false
+    Boolean downsample = false
+    Boolean sample_sites = false
+    String? read_cleaner = 'CGP'
+    String? mapper
+    String? snpcaller
+  }
+  command <<<
+    date | tee DATE
+
+    # set bash arrays based on inputs to ensure read arrays are of equal length
+    read1_array=(~{sep=' ' read1})
+    read1_array_len=$(echo "${#read1[@]}")
+    read2_array=(~{sep=' ' read2})
+    read2_array_len=$(echo "${#read2[@]}")
+
+    if [ "$read1_array_len" -ne "$read2_index_array_len" ]; then
+      echo "read1 array (length: $read1_array_len) and read2 index array (length: $read2_array_len) are of unequal length." >&2
+      exit 1
+    fi
+
+    # create lyvset project
+    set_manage.pl --create ~{dataset_name}
+
+    #shuffle paired end reads
+    for index in ${!read1_array[@]}; do
+      cp ${read1_array[$index]} . && cp ${read2_array[$index]} . # move reads to cwd
+    done
+
+    shuffleSplitReads.pl --numcpus ~{cpu} -o ./interleaved *.fastq.gz 
```

Comment:
I think we are OK to leave this, but this line requires that input FASTQ files end in `.fastq.gz`. Just something to be aware of in case anyone names their FASTQ files with `.fq.gz` or something like that and the workflow fails